### PR TITLE
[sources] WITH TIMELINE

### PR DESCRIPTION
**Pull request by bkirwi**
Moves the `TIMELINE` param option to the new WITH block, adding a keyword etc.

### Motivation

Known valuable: https://github.com/MaterializeInc/materialize/issues/14341 and https://github.com/MaterializeInc/materialize/issues/13919

### Checklist

- [ ] This PR has adequate test coverage / QA involvement has been duly considered.
- [ ] This PR evolves [an existing `$T ⇔ Proto$T` mapping](https://github.com/MaterializeInc/materialize/blob/main/doc/developer/command-and-response-binary-encoding.md) (possibly in a backwards-incompatible way) and therefore is tagged with a `T-protobuf` label.
- [ ] This PR includes the following [user-facing behavior changes](https://github.com/MaterializeInc/materialize/blob/main/doc/developer/guide-changes.md#what-changes-require-a-release-note):
  - Changes parsing behaviour behind the unsafe flag, IIUC.

### CODE REVIEW

File: src/sql/src/plan/statement/ddl.rs [hunk 1]
```diff
@@ -3761,27 +3766,33 @@ pub fn plan_alter_source(
...
+                remote = AlterSourceItem::Set(value);
+            }
             if let Some(value) = size_opt {
                 size = AlterSourceItem::Set(value);
             }
-
-            if let Some(value) = remote_opt {
-                remote = AlterSourceItem::Set(value);
+            if let Some(_) = timeline_opt {
+                sql_bail!("Cannot modify the TIMELINE of a SOURCE.");
```
Comment (guswynn) [inline src/sql/src/plan/statement/ddl.rs:3782]: unfortunate that we have to do this here, perhaps we can add a `TODO: split CreateSourceOptionExtracted into alterable and non-alterable structs`

File: src/sql/src/plan/statement/ddl.rs [hunk 2]
```diff
@@ -857,12 +866,8 @@ pub fn plan_create_source(
...
 
     // Allow users to specify a timeline. If they do not, determine a default timeline for the source.
-    let timeline = if let Some(timeline) = legacy_with_options.remove("timeline") {
-        match timeline.into() {
-            Some(Value::String(timeline)) => Timeline::User(timeline),
-            Some(v) => sql_bail!("unsupported timeline value {}", v.to_ast_string()),
-            None => sql_bail!("unsupported timeline value: secret"),
-        }
+    let timeline = if let Some(timeline) = timeline {
+        Timeline::User(timeline)
```
Comment (guswynn) [inline src/sql/src/plan/statement/ddl.rs:870]: this needs to `require_unsafe_mode` as `legacy_options` already require unsafe mode

File: .editorconfig [hunk 3]
```diff
@@ -19,6 +19,9 @@ indent_size = 2
 [*.jsx]
 indent_size = 2
 
+[*.td]
+indent_size = 2
+
```
Comment (guswynn) [inline .editorconfig:24]: personally i would break this commit out to a different pr...ill stamp that one tho!

**Review by guswynn — APPROVED** [review]
looks good minus a few nits + needing to require unsafe mode

accepting to merge after those are resolved!

File: src/sql/src/plan/statement/ddl.rs (hunk 2 above)
Comment (bkirwi) [reply src/sql/src/plan/statement/ddl.rs:870]: Already does! https://github.com/MaterializeInc/materialize/blob/e121eda2d7cca47530db6887c29e69c5ba28d47f/src/sql/src/plan/statement/ddl.rs#L336

Comment (guswynn) [reply src/sql/src/plan/statement/ddl.rs:870]: oh wait this is covered by the `SAFE_WITH_OPTIONS` check above, ignore me!

Comment (guswynn) [reply src/sql/src/plan/statement/ddl.rs:870]: oh press send on github at the same time! nice!!

File: src/sql/src/plan/statement/ddl.rs (hunk 1 above)
Comment (bkirwi) [reply src/sql/src/plan/statement/ddl.rs:3782]: _shrug_... sort of unfortunate, but i'm hesitant to generalize from one example! (I think that may require changes to the macro, and it makes it harder to validate whether all options are consumed.)

File: .editorconfig (hunk 3 above)
Comment (bkirwi) [reply .editorconfig]: Reverted!

File: src/sql/src/plan/statement/ddl.rs (hunk 1 above)
Comment (guswynn) [reply src/sql/src/plan/statement/ddl.rs:3782]: there will be at least 2 more examples of this, `epoch_timestamp_ms`, and `TIMESTAMP GRANULARITY`

Comment (guswynn) [reply src/sql/src/plan/statement/ddl.rs:3782]: we can add the todo when we add those
